docs/cli-reference/dfx-extension.md

```diff
@@ -0,0 +1,83 @@
+# dfx extension
+
+Use the `dfx extension` command to manage the extensions available in the `dfx` tool. Extensions can provide additional functionality and commands to the `dfx` tool, enhancing its capabilities.
+
+## Basic Usage
+
+```bash
+dfx extension [command] [option] [flag]
+```
+
+## Commands
+
+| Command                                  | Description                     |
+|------------------------------------------|---------------------------------|
+| [`install`](#dfx-extension-install)      | Installs an extension.          |
+| [`uninstall`](#dfx-extension-uninstall)  | Uninstalls an extension.        |
+| [`run`](#dfx-extension-run)              | Executes an extension.          |
+| [`list`](#dfx-extension-list)            | Lists all installed extensions. |
+
+
+## dfx extension install
+
+**Installing an Extension**:
+
+```bash
+dfx extension install sns
+```
+
+This command will install the `sns` extension.
```

Comment:
Please describe what happens if the extension is already installed.  (note: I have filed https://dfinity.atlassian.net/browse/SDK-1309 to make `extension install` behavior consistent with things like `apt-get install` and `brew install`).